### ISSUE TITLE
On .K, .Map, .List, .KList ...

### ISSUE THREAD

**Issue by grosu** [kframework/k-legacy#1772]
... and other such qualified dots: THERE SHOULD BE NONE OF THEM ANYMORE!  I see that imp++ uses `.K`.  Why?  All dots are polymorphic now, and we should just cast them if the inferencer is not smart enough: `.::K`, `.::Klist`, etc.  I'm marking this is as urgent bug, because we should six it quickly.  Who introduced the qualified dots?  I think that person should volunteer to change them all to `.`.  If nobody volunteers, then @radumereuta , it falls into your territory, so please fix it asap.

**Comment by bmmoore** [comment]
We also still have `.Bag` being used for the absence of cells.

**Comment by radumereuta** [comment]
In `kast.k` we have something like this:

```
syntax KBott ::= ".K"      [klabel(#EmptyK), hook(org.kframework.kore.EmptyK)]
               | ".::K"    [klabel(#EmptyK), hook(org.kframework.kore.EmptyK)]
```

This means we can have both the old `.K` notation, as well as the new qualified dot.
I chose to leave them both so we have an easier time translating definitions from K3.6 to K4 (KORE).

I can remove the old format, but are we ready to remove SDF? Because that's the only way I can make sure we don't need the old notation.
Having that much old code is kind of hindering progress. Is the new implementation stable enough to be used in the RV projects? Because that was the reason a few months ago for delaying the removal of SDF.

**Comment by grosu** [comment]
@radumereuta , again: we should have _polymorphic_ `.` only.  That is, both your productions

```
syntax KBott ::= ".K"      [klabel(#EmptyK), hook(org.kframework.kore.EmptyK)]
               | ".::K"    [klabel(#EmptyK), hook(org.kframework.kore.EmptyK)]
```

are wrong.  You should only have `.`, and then to allow `.::K` in ambiguous situations as an instance of the general casting notation.

**Comment by radumereuta** [comment]
Yes, that is a hack for now, because the disambiguation algorithm for polymorphic productions is not yet clear to me. I will get to it when the problem is more simple.
You haven't answered my question about SDF.

**Comment by grosu** [comment]
I am OK with getting rid of SDF, so I'll let others answer.  I think the original idea was to have some of the examples in the tutorial work, so we have tests to make sure we do not break other stuff.  Now we do have some tests.  @dwightguth , would the removal of SDF break C?

> Yes, that is a hack for now...

That's the problem which slowed us down for so long, especially you.  Always hacks and patches.  Why don't you try something different for a change: stop and spend some time understanding the problem at its root, and then implement it.  In what regards me, I think all the details needed to understand this problem have been discussed in previous posts and messages.  If you have specific questions, ask them.  Point is, we have to do this properly once and for all, without hacks.

**Comment by dwightguth** [comment]
Removing SDF should not be an issue for C.

**Comment by grosu** [comment]
@cos, @andreistefanescu : can you please get rid of SDF asap (sorry if you've done it already, I was a bit disconnected for the last couple of weeks)?

**Comment by radumereuta** [comment]
SDF is out of master.
